File: Array/leetcode_3132.cpp

// Problem Link ====>>https://leetcode.com/problems/find-the-integer-added-to-array-ii/description/

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int minimumAddedInteger(vector<int> &nums1, vector<int> &nums2)
    {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        int ans = INT_MAX;
        for (int i = 0; i < nums1.size(); i++)
        {
            for (int j = i + 1; j < nums1.size(); j++)
            {
                vector<int> updatedNums1;
                for (int k = 0; k < nums1.size(); k++)
                {
                    if (k == i || k == j)
                        continue;
                    updatedNums1.push_back(nums1[k]);
                }
                int flag = 1;
                int diff = nums2[0] - updatedNums1[0];
                for (int k = 1; k < updatedNums1.size(); k++)
                {
                    if (nums2[k] - updatedNums1[k] != diff)
                        flag = 0;
                }
                if (flag)
                    ans = min(ans, nums2[0] - updatedNums1[0]);
            }
        }
        return ans;
    }
};

int main()
{
    Solution s;
    int n;
    cin >> n;
    vector<int> nums1;
    vector<int> nums2;
    cout << "Enter elements for the first array:\n";
    for (int i = 0; i < n; ++i)
    {
        cin >> nums1[i];
    }

    cout << "Enter elements for the second array:\n";
    for (int j = 0; j < n; ++j)
    {
        cin >> nums2[j];
    }

    int x = s.minimumAddedInteger(nums1, nums2);

    cout << x << endl;
    return 0;
    return 0;
}

File: Array/leetcode_3132.py

# Problem Link ====>>https://leetcode.com/problems/find-the-integer-added-to-array-ii/description/

class Solution:
    def minimumAddedInteger(self, nums1: list[int], nums2: list[int]) -> int:
        nums1.sort()
        nums2.sort()
        diff = nums2[0] - nums1[0]

        for i in range(len(nums1)):
            if nums2[i] - nums1[i] != diff:
                new_diff = nums2[i] - nums1[i]

                can_achieve = any(num + new_diff >= nums2[0] for num in nums1)

                if not can_achieve:
                    return INT_MAX

        return diff


if __name__ == "__main__":
    solution = Solution()

    n = int(input("Enter the size of both arrays: "))

    nums1 = []
    nums2 = []

    print("Enter elements for the first array:")
    for i in range(n):
        num = int(input())
        nums1.append(num)

    print("Enter elements for the second array:")
    for i in range(n):
        num = int(input())
        nums2.append(num)

    min_added_integer = solution.minimumAddedInteger(nums1, nums2)

    if min_added_integer == INT_MAX:
        print("No such integer exists.")
    else:
        print("Minimum integer that needs to be added:", min_added_integer)
